lib/project.ml

```diff
@@ -4,20 +4,47 @@ type t = Fpath.t
 
 let folder_blacklist = [ "_build"; "_opam"; Fpath.to_string Config.vendor_dir ]
 
+let repo_filename = "repo"
+
+let packages_folder = "packages"
+
+let is_repo_file path =
+  let basename = Fpath.to_string (Fpath.base path) in
+  if String.equal basename repo_filename then Bos.OS.File.exists path
+  else Ok false
+
+let is_packages_folder path =
+  let basename = Fpath.to_string (Fpath.base path) in
+  if String.equal basename packages_folder then Bos.OS.Dir.exists path
+  else Ok false
+
+let is_opam_repo dir =
+  let open Result.O in
+  let* is_dir = Bos.OS.Dir.exists dir in
+  if not is_dir then Ok false
+  else
+    let* content = Bos.OS.Dir.contents dir in
+    let* has_repo_file = Result.List.exists content ~f:is_repo_file in
+    let* has_pkg_folder = Result.List.exists content ~f:is_packages_folder in
+    Ok (has_repo_file && has_pkg_folder)
+
+let is_blacklisted path =
```

Comment:
We should probably not use the term blacklist in new code.

Reply:
Why is that?

Reply:
Similar as `master` vs `main`. There have been a lot of writing about it lately, here's [one article](https://romanisthere.github.io/posts/blacklist/), [some more](https://www.adexchanger.com/data-driven-thinking/no-more-inflammatory-jargon-change-blacklist-to-blocklist/) and [yet some more](https://www.vice.com/en/article/v7dd3d/we-need-to-stop-saying-blacklist-and-whitelist).

While just renaming things for the sake of renaming creates headaches like the `main`-`master` split we have in git I usually try to not introduce problematic terms in new code.